lua/octo/workflow_runs.lua

```diff
@@ -723,4 +736,39 @@ M.refetch = function()
   populate_preview_buffer(id, M.buf)
 end
 
+---@param db_id number | nil
+M.cancel = function(db_id)
+  local id = db_id or M.current_wf.databaseId
+  local _, stderr = gh.run.cancel {
+    id,
+    opts = { mode = "sync" },
+  }
+  if stderr and not utils.is_blank(stderr) then
+    vim.api.nvim_err_writeln(stderr)
+    utils.error "Failed to cancel workflow run"
+  else
+    utils.info "Cancelled"
+  end
+  M.refetch()
+end
+
+---@param opts { db_id: number | nil, failed: boolean | nil }
+M.rerun = function(opts)
+  opts = opts or {}
+  local failed_jobs = opts.failed == true
+  local id = opts.db_id or (M.current_wf and M.current_wf.databaseId)
+  local _, stderr = gh.run.rerun {
+    id,
+    failed_jobs and "--failed" or "",
```

Comment:
I think this works. intended use was with a table { failed = true / false } which get translated to --failed. I'm not sure out the empty string behavior

{ "" } would lead to an empty string in plenary.Job table 

You haven't had an issue?

Reply:
Havent had an issue but will refactor. True false approach is cleaner. I didnt know it was supported

Reply:
Refactored